src/GenAPI/Microsoft.DotNet.GenAPI/GenAPIApp.cs

```diff
@@ -0,0 +1,185 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiSymbolExtensions;
+
+namespace Microsoft.DotNet.GenAPI
+{
+    /// <summary>
+    /// Class to standertize initilization and running of GenAPI tool.
+    ///     Shared between CLI and MSBuild tasks frontends.
+    /// </summary>
+    public static class GenAPIApp
+    {
+        public sealed class Context
```

Comment:
I don't think that this extra class is necessary, I would just pass all these in as parameters to GenAPIApp but if you want to keep it, make it a readonly struct instead (for better perf).